add basic cirq and qualtran tutorial

qqUAnH opened the pull request:
I've added tutorials for Cirq and Qualtran. These tutorials were developed for the course "Practical Quantum Computing" by me and Professor Alexandru Paler, and they've received positive feedback from our students.

mpharrigan commented:
Thanks for sharing your tutorial notebook! It looks like a nice way of tying together a lot of the documentation floating around and is made interactive with the exercise portions. 

In its current form, I don't think it's completely appropriate for wholesale inclusion in the qualtran docs. Here are some major organizational changes if you'd like to contribute this to the main qualtran docs

 - the Cirq tutorial is out of the scope for this project
 - For the parts based on existing documentation: I'd like to keep this as in-sync as possible. So if you made any changes or pedagogical improvements -- please contribute them "upstream" to the original documentation notebooks (hosted in this repository).
 - Then: it is important to link from this all-in-one tutorial back to the topic-based documentation on which it is based. Basically, we need this tutorial to fit into a cohesive doc suite

-----

If you're happy with the tutorial as written, an alternative would be for you to host it yourself and we can link out to it from the docs

qqUAnH commented:
Hi @mpharrigan, I have removed Cirq from the all-in-one tutorial and updated it with references and a proper introduction. I also made some small modification in the original Bloq-tutorial and Cirq Interoperability notebook to keep everything in-sync

mpharrigan commented on qualtran/_infra/Bloqs-Tutorial.ipynb:
```diff
@@ -83,29 +109,46 @@
    "id": "22d8749d",
-   "metadata": {},
-   "outputs": [],
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-06-11T03:34:54.004785Z",
+     "start_time": "2024-06-11T03:34:51.338589Z"
+    },
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 351
+    },
+    "id": "22d8749d",
+    "outputId": "b4764180-e624-4579-9e94-f7dc277f374b"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Register(name='control', dtype=QBit(), _shape=(), side=<Side.THRU: 3>)"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
```
can you "clear output of all cells" before committing these notebooks so the diff is clearer and we don't clutter the git history

qqUAnH replied:
Thanks. I will do that

qqUAnH replied:
@mpharrigan I have removed output of all cell and the outdated table of contents in all-in-one tutorial.
